osmaxx/profile/views.py

```diff
@@ -1,20 +1,148 @@
+from django.conf import settings
 from django.contrib import messages
-from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.cache import cache
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.mail import send_mail
 from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.template.defaultfilters import urlencode
+from django.utils.translation import ugettext as _
 from django.views import generic
 
 from osmaxx.contrib.auth.frontend_permissions import LoginRequiredMixin
 from osmaxx.profile.forms import ProfileForm
+from osmaxx.profile.models import Profile
 
 
-class ProfileView(LoginRequiredMixin, generic.UpdateView):
-    model = get_user_model()
+class SendVerificationEmailMixin(object):
+    RATE_LIMIT_SECONDS = 30
+
+    def _send_email_verification(self, profile):
+        if cache.get(profile.associated_user.id):
+            return
+        to_email = profile.unverified_email
+        if to_email:
+            cache.set(profile.associated_user.id, 'dummy value', timeout=self.RATE_LIMIT_SECONDS)
+            user_administrator_email = settings.OSMAXX['ACCOUNT_MANAGER_EMAIL']
+            token = profile.activation_key()
+            subject = _('Verify your email address')
+            message = _('In order to verify your email-address click the following link:\n'
+                        '{}?token={}'
+                        '\n'
+                        'If it wasn\'t you, just ignore this email.\n'
```

Comment:
putting both in a template would be a good idea, it's much more customizable. Good idea :+1: